test-utils/src/data_gen.rs

```diff
@@ -169,7 +179,19 @@ impl BatchBuilder {
             .append_option(rng.gen_bool(0.9).then(|| rng.gen()));
         self.response_status
             .append_value(status[rng.gen_range(0..status.len())]);
-        self.prices_status.append_value(self.row_count as i128);
+        self.decimal128_price.append_value(self.row_count as i128);
+
+        let val = rng.gen_range(0..100);
```

Comment:
I think the fact that this code picks a new random number is why all the values in the test need to change

I wonder if you could reuse the same value that was created for some other field to avoid havint od this

For example, maybe you could reuse the value of the first ip byte

```rust
       let first_ip_byte = rng.gen::<u8>()
        self.client_addr.append_value(format!(
            "{}.{}.{}.{}",
            first_ip_byte,
            rng.gen::<u8>(),
            rng.gen::<u8>(),
            rng.gen::<u8>()
        ));
...
        // Avoid row group min_max to prune this, so skip 50 between 0..99
        if first_ip_byte == 50 {
            self.decimal128_id_int32.append_value(1);
            self.decimal128_id_int64.append_value(1);
            self.decimal128_id_fixed_len_byte.append_value(1);
        } else {
            self.decimal128_id_int32.append_value(first_ip_byte);
            self.decimal128_id_int64.append_value(first_ip_byte);
            self.decimal128_id_fixed_len_byte.append_value(first_ip_byte);
        }
...